src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift

```diff
@@ -267,6 +273,45 @@ class ExposureSubmissionCoordinator: NSObject, ExposureSubmissionCoordinating, R
 		)
 	}
 
+	private func createWarnOthersViewController() -> ExposureSubmissionWarnOthersViewController {
+		var vc: ExposureSubmissionWarnOthersViewController?
+
+		// ugly but works for the moment
+		// refactoring more of the coordinator-logic to facilitate combine would help
+		let group = DispatchGroup()
+		group.enter()
+
+		// get list of supported countries and create warn others view controller
+		appConfigurationProvider
+			.supportedCountries()
+			.sink(
+				receiveCompletion: { _ in
+					// no op
+				},
+				receiveValue: { countries in
+					vc = ExposureSubmissionWarnOthersViewController(
+						supportedCountries: countries,
+						onPrimaryButtonTap: { [weak self] isLoading in
+							self?.model.exposureSubmissionService.isSubmissionConsentGiven = true
+							self?.model.exposureSubmissionService.getTemporaryExposureKeys { error in
+								isLoading(false)
+								if let error = error {
+									self?.showErrorAlert(for: error)
+								} else {
+									self?.showThankYouScreen()
+								}
+							}
+						}
+					)
+					group.leave()
```

Comment:
This is perfect for the completion closure.

Reply:
This is what `getInitialViewController` unfortunately expects. I did not come up with this mess and just copy the general behavior until we refactor the coordinator.

Reply:
I don't get it.
Whats the problem with moving `group.leave()` to completion closure of the sink?

Reply:
It made no difference in this situation. But it's moved now.

Reply:
Still cleaner and safer for changes. Thanks